src/poetry/core/masonry/builders/builder.py

```diff
@@ -378,14 +376,11 @@ def __init__(
 
         self.path = self.path.resolve()
 
-    def __eq__(self, other: BuildIncludeFile | Path) -> bool:
-        if hasattr(other, "path"):
-            return self.path == other.path
-
-        return self.path == other
+    def __eq__(self, other: object) -> bool:
```

Comment:
Why are we disallowing comparison with Path?

Reply:
Not disallowing, but always returning False.

Because a BuildIncludeFile is not a Path!